### PR TITLE
Fix some spotbugs issues

### PR DESCRIPTION
**Pull request by Siedlerchr**
Some few issues, mostly some issues with equal:
https://spotbugs.github.io/

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Screenshots added (for bigger UI changes)
- [ ] Manually tested changed features in running JabRef
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

- [ ] If you changed the localization: Did you run `gradle localizationUpdate`?

### CODE REVIEW

File: src/main/java/org/jabref/Globals.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ private Globals() {
     }
 
     // Key binding preferences
-    public static KeyBindingRepository getKeyPrefs() {
+    public static synchronized KeyBindingRepository getKeyPrefs() {
```
Comment (stefan-kolb) [inline src/main/java/org/jabref/Globals.java]: Why synchronized here?

File: src/main/java/org/jabref/gui/BasePanel.java [hunk 2]
```diff
@@ -2144,7 +2144,9 @@ public void action() {
                 if ((focused != null) && (focused instanceof FieldEditor) && focused.hasFocus()) {
                     // User is currently editing a field:
                     // Check if it is the preamble:
-                    if ((preambleEditor != null) && (focused == preambleEditor.getFieldEditor())) {
+
+                    FieldEditor f = (FieldEditor) focused;
```
Comment (stefan-kolb) [inline src/main/java/org/jabref/gui/BasePanel.java]: Bad variable name

File: src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java [hunk 3]
```diff
@@ -150,8 +150,8 @@ public void openFiles(List<Path> filesToOpen, boolean raisePanel) {
             Path file = iterator.next();
             for (int i = 0; i < frame.getTabbedPane().getTabCount(); i++) {
                 BasePanel basePanel = frame.getBasePanelAt(i);
-                if ((basePanel.getBibDatabaseContext().getDatabaseFile().isPresent())
-                        && basePanel.getBibDatabaseContext().getDatabaseFile().get().equals(file)) {
+                if ((basePanel.getBibDatabaseContext().getDatabasePath().isPresent())
```
Comment (stefan-kolb) [inline src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java]: Path equals file?

File: src/main/java/org/jabref/logic/citationstyle/CitationStyle.java [hunk 4]
```diff
@@ -163,16 +163,22 @@ public String toString() {
     }
 
     @Override
-    public boolean equals(Object other) {
-        if (this == other) {
+    public boolean equals(Object o) {
```
Comment (stefan-kolb) [inline src/main/java/org/jabref/logic/citationstyle/CitationStyle.java]: other is a pretty good name for the parameter imho

File: src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java (hunk 3 above)
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/importer/actions/OpenDatabaseAction.java]: This was the bug I fixed

File: src/main/java/org/jabref/Globals.java (hunk 1 above)
Comment (Siedlerchr) [inline src/main/java/org/jabref/Globals.java]: Incorrect lazy initialization 
https://stackoverflow.com/a/6782690/3450689

Comment (koppor) [inline src/main/java/org/jabref/Globals.java]: Refs [item 3 of effective java](https://github.com/mgp/book-notes/blob/master/effective-java-2nd-edition.markdown#item-3-enforce-the-singleton-property-with-a-private-constructor-or-an-enum-type), doesn't it?

File: src/main/java/org/jabref/logic/citationstyle/CitationStyle.java (hunk 4 above)
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/citationstyle/CitationStyle.java]: I renamed it to be consistent with the other implementations we have.
And there other is used as the name of the casted instance, see the line below

**Pull request merged**